### ISSUE TITLE
Compiler does not recognize symbol names in destructuring "let" assignments

### ISSUE THREAD

**Issue by cboulanger** [qooxdoo/qooxdoo-compiler#120]
Given this code
```javascript
 let childControls = [
      { id : "main" },
      { id : "extended" },
      { id : "toc" }
    ];
    for( let {id} of childControls ){
      let methodName = 'create' + id.charAt(0).toUpperCase() + id.substr(1) + "View";
      let [ page, button ] = this[methodName]();
      stackView.add(page);
      menuBar.add(button);
    }
```

I get the following warning from the compiler:
```
[…] Unresolved use of symbol id.charAt
[…] Unresolved use of symbol page.setWidgetId
```

The  the problem is that the `for( let {id} of childControls )` is not detecting the declaration of the `id` variable and also that `let [ page, button ]` is not detecting the declaration of `page`.
